R/tools.R

```diff
@@ -21,13 +21,21 @@
 #' \dontrun{
 #' ch <- ellmer::chat_anthropic()
 #'
-#' btw_register_tools(ch)
+#' # register all of the available tools
+#' ch$set_tools(btw_tools())
+#' 
+#' # or register only the tools related to fetching documentation
+#' ch$set_tools(btw_tools(tools = "docs"))
 #' }
 #'
 #' @family Tools
 #' @export
-btw_register_tools <- function(chat, tools = NULL) {
-  check_inherits(chat, "Chat")
+btw_tools <- function(tools = NULL) {
+  has_tools <- !is.null(tools)
+
+  if (!has_tools) {
+    return(as_ellmer_tools(.btw_tools))
+  }
```

Comment:
Doesn't look like we use `has_tools` anywhere else

```suggestion
  if (is.null(tools)) {
    return(as_ellmer_tools(.btw_tools))
  }
```